src/config/src/yaml.rs

```diff
@@ -113,6 +113,21 @@ pub trait Element: Sized {
         None
     }
 
+    /// From some string (should be used for top level elements if you want to embed the code)
+    fn from_string<'a>(src: &'a str) -> Result<Self, ConfigError> {
```

Comment:
Do you need the explicit lifetime, tho? You're not returning a reference, so there seems to not be any use for it